appscale/tools/agents/azure_agent.py

```diff
@@ -262,25 +307,59 @@ def run_instances(self, count, parameters, security_configured):
       public_ips: A list of public IP addresses.
       private_ips: A list of private IP addresses.
     """
+    is_verbose = parameters[self.PARAM_VERBOSE]
     credentials = self.open_connection(parameters)
     resource_group = parameters[self.PARAM_RESOURCE_GROUP]
     subscription_id = parameters[self.PARAM_SUBSCRIBER_ID]
     network_client = NetworkManagementClient(credentials, subscription_id)
     virtual_network = parameters[self.PARAM_GROUP]
     subnet = self.create_virtual_network(network_client, parameters,
                                          virtual_network, virtual_network)
+
+    handles = []
     for _ in range(count):
-      vm_network_name = Haikunator().haikunate()
-      self.create_network_interface(network_client, vm_network_name,
-        vm_network_name, subnet, parameters)
-      network_interface = network_client.network_interfaces.get(
-        resource_group, vm_network_name)
-      self.create_virtual_machine(credentials, network_client,
-        network_interface.id, parameters, vm_network_name)
+      handles.append(self.create_vm_bundle_async(
+        credentials, subscription_id, subnet, parameters, resource_group))
+    for handle in handles:
+       handle.join()
 
     public_ips, private_ips, instance_ids = self.describe_instances(parameters)
+    AppScaleLogger.verbose("Instance Info:\n{}\n{}\n{}".format(public_ips,
+                           private_ips, instance_ids), is_verbose)
     return instance_ids, public_ips, private_ips
 
+
+  def create_vm_bundle(self, credentials, subscription_id, subnet, parameters,
+                       resource_group):
+    """ Creates the necessary resources for a Virtual Machine (Public IP
+    Address, Network Interface and Virtual Machine).
+
+    Args:
+      credentials: A ServicePrincipalCredentials object.
+      subscription_id: A string, the subscription ID.
+      subnet: A virtual network subnet.
+      parameters: A dictionary of parameters necessary to create resources.
+      resource_group: An Azure resource group.
+    """
+    network_client = NetworkManagementClient(credentials, subscription_id)
+    vm_network_name = Haikunator().haikunate()
+
+    self.create_network_interface(network_client, vm_network_name,
+      vm_network_name, subnet, parameters)
+    try:
+      network_interface = network_client.network_interfaces.get(
+        resource_group, vm_network_name)
+    except CloudError:
+      AppScaleLogger.warn("Error: Network Interface: {} not found. "
+                          "Skipping...".format(vm_network_name))
+      return
+
+    self.create_virtual_machine(credentials, network_client,
+      network_interface.id, parameters, vm_network_name)
+    time.sleep(2 * self.SLEEP_TIME)
+    self.RI_CACHE.put(vm_network_name)
```

Comment:
@cdonati a resource identifier name is a string and is the same for VMs, networks, nics and IPs.